docs/slotfilling.rst

```diff
@@ -3,12 +3,284 @@
+                                      "that it has to fill")
+
+        def slot_mappings(self):
+            # type: () -> Dict[Text: Union[Dict, List[Dict]]]
+            """A dictionary to map required slots to
+                - an extracted entity
+                - intent: value pairs
+                - a whole message
+                or a list of them, where the first match will be picked
+                Empty dict is converted to a mapping of
+                the slot to the extracted entity with the same name
+            """
+
+            return {}
+
+
+        def validate(self, dispatcher, tracker, domain):
+              # type: (CollectingDispatcher, Tracker, Dict[Text, Any]) -> List[Dict]
+              """Validate extracted value of requested slot
+                  else reject execution of the form action
+                  Subclass this method to add custom validation and rejection logic
+              """
+              # extract other slots that were not requested
+              # but set by corresponding entity
+              slot_values = self.extract_other_slots(dispatcher, tracker, domain)
+
+              # extract requested slot
+              slot_to_fill = tracker.get_slot(REQUESTED_SLOT)
+              if slot_to_fill:
+                  slot_values.update(self.extract_requested_slot(dispatcher,
+                                                                 tracker, domain))
+                  if not slot_values:
+                      # reject to execute the form action
+                      # if some slot was requested but nothing was extracted
+                      # it will allow other policies to predict another action
+                      raise ActionExecutionRejection(self.name(),
+                                                     "Failed to validate slot {0} "
+                                                     "with action {1}"
+                                                     "".format(slot_to_fill,
+                                                               self.name()))
+
+              # validation succeed, set slots to extracted values
+              return [SlotSet(slot, value) for slot, value in slot_values.items()]
+
+          def submit(self, dispatcher, tracker, domain):
+              # type: (CollectingDispatcher, Tracker, Dict[Text, Any]) -> List[Dict]
+              """Define what the form has to do
+                  after all required slots are filled"""
+
+              raise NotImplementedError("A form must implement a submit method")
+
+
+For the RestaurantForm, you first define it's name and the list of required slots:
+
+.. code-block:: python
+
+ class RestaurantForm(FormAction):
+    """Example of a custom form action"""
+
+    def name(self):
+        # type: () -> Text
+        """Unique identifier of the form"""
+
+        return "restaurant_form"
+
+    @staticmethod
+    def required_slots():
```

Comment:
Did we decide against using the `tracker` as an argument?  In my implementation, this was the most elegant solution for auxiliary slots

Reply:
no we did not i just did not update that function in the docs

Reply:
also please hit the start a review button rather than posting individual comments 😄